cmd/juju/common/model.go

```diff
@@ -72,35 +72,52 @@ func ModelInfoFromParams(info params.ModelInfo, now time.Time) (ModelInfo, error
 	if err != nil {
 		return ModelInfo{}, errors.Trace(err)
 	}
-	status := ModelStatus{
-		Current: info.Status.Status,
-		Message: info.Status.Info,
-		Since:   friendlyDuration(info.Status.Since, now),
-	}
-	if info.Migration != nil {
-		status.Migration = info.Migration.Status
-		status.MigrationStart = friendlyDuration(info.Migration.Start, now)
-		status.MigrationEnd = friendlyDuration(info.Migration.End, now)
-	}
 	cloudTag, err := names.ParseCloudTag(info.CloudTag)
 	if err != nil {
 		return ModelInfo{}, errors.Trace(err)
 	}
-	return ModelInfo{
+	modelInfo := ModelInfo{
 		Name:           info.Name,
 		UUID:           info.UUID,
 		ControllerUUID: info.ControllerUUID,
 		Owner:          tag.Id(),
 		Life:           string(info.Life),
-		Status:         status,
 		Cloud:          cloudTag.Id(),
 		CloudRegion:    info.CloudRegion,
-		ProviderType:   info.ProviderType,
-		Users:          ModelUserInfoFromParams(info.Users, now),
-		Machines:       ModelMachineInfoFromParams(info.Machines),
-		SLA:            modelSLAFromParams(info.SLA),
-		SLAOwner:       modelSLAOwnerFromParams(info.SLA),
-	}, nil
+	}
+	if info.Status != nil {
```

Comment:
To avoid using a pointer and breaking other code, this would need to be
if info.Status != params.EntityStatus{}